x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/loadingservice/ModelLoadingService.java

```diff
@@ -77,13 +125,15 @@ public void getModel(String modelId, ActionListener<Model> modelActionListener)
      */
```

Comment:
Not really, if `getModel(...)` sees that somebody else kicked of a load (cluster state change?) but it is not loaded yet, this method will add the listener to the listener queue. 

If the caller sees that the model is not in the middle of loading AND is not loaded, it will make a direct call to eager load it and NOT attempt to cache it. 

I will try to clarify the comment.